stacker_blueprints/eks.py

```diff
@@ -0,0 +1,427 @@
+        if role_arn:
+            return role_arn
+        return self.create_iam_role()
+
+    def create_template(self):
+        t = self.template
+        role_arn = self.get_iam_role()
+        variables = self.get_variables()
+
+        args = {}
+        version = variables["Version"]
+        if version:
+            args["Version"] = version
+
+        # This is a fully qualified stacker name, prefixed with the namespace.
+        eks_name_tag = self.context.get_fqn(variables["Name"])
+
+        t.add_resource(
+            eks.Cluster(
+                "EksCluster",
+                Name=eks_name_tag,
+                RoleArn=role_arn,
+                ResourcesVpcConfig=eks.ResourcesVpcConfig(
+                    SecurityGroupIds=variables["SecurityGroupIds"].split(","),
+                    SubnetIds=variables["SubnetIds"].split(","),
+                ),
+                **args
+            )
+        )
+
+        # Output the ClusterName and RoleArn, which are useful as inputs for
+        # EKS worker nodes.
+        t.add_output(Output("ClusterName", Value=eks_name_tag))
+        t.add_output(Output("RoleArn", Value=role_arn))
+
+
+# This is a stacker re-implementation of Amazon's template defined here:
+# https://amazon-eks.s3-us-west-2.amazonaws.com/1.10.3/2018-06-05/amazon-eks-nodegroup.yaml
+# More docs can be found here:
+# https://docs.aws.amazon.com/eks/latest/userguide/launch-workers.html
+
+# This comes straight from that template above, just formatted for Python
+max_pods_per_instance = {
+    "c4.large": 29,
+    "c4.xlarge": 58,
+    "c4.2xlarge": 58,
+    "c4.4xlarge": 234,
+    "c4.8xlarge": 234,
+    "c5.large": 29,
+    "c5.xlarge": 58,
+    "c5.2xlarge": 58,
+    "c5.4xlarge": 234,
+    "c5.9xlarge": 234,
+    "c5.18xlarge": 737,
+    "i3.large": 29,
+    "i3.xlarge": 58,
+    "i3.2xlarge": 58,
+    "i3.4xlarge": 234,
+    "i3.8xlarge": 234,
+    "i3.16xlarge": 737,
+    "m3.medium": 12,
+    "m3.large": 29,
+    "m3.xlarge": 58,
+    "m3.2xlarge": 118,
+    "m4.large": 20,
+    "m4.xlarge": 58,
+    "m4.2xlarge": 58,
+    "m4.4xlarge": 234,
+    "m4.10xlarge": 234,
+    "m5.large": 29,
+    "m5.xlarge": 58,
+    "m5.2xlarge": 58,
+    "m5.4xlarge": 234,
+    "m5.12xlarge": 234,
+    "m5.24xlarge": 737,
+    "p2.xlarge": 58,
+    "p2.8xlarge": 234,
+    "p2.16xlarge": 234,
+    "p3.2xlarge": 58,
+    "p3.8xlarge": 234,
+    "p3.16xlarge": 234,
+    "r3.xlarge": 58,
+    "r3.2xlarge": 58,
+    "r3.4xlarge": 234,
+    "r3.8xlarge": 234,
+    "r4.large": 29,
+    "r4.xlarge": 58,
+    "r4.2xlarge": 58,
+    "r4.4xlarge": 234,
+    "r4.8xlarge": 234,
+    "r4.16xlarge": 737,
+    "t2.small": 8,
+    "t2.medium": 17,
+    "t2.large": 35,
+    "t2.xlarge": 44,
+    "t2.2xlarge": 44,
+    "x1.16xlarge": 234,
+    "x1.32xlarge": 234,
+}
+
+
+def create_max_pods_per_node_mapping(t):
+    mapping = {}
+    for instance, max_pods in max_pods_per_instance.iteritems():
```

Comment:
nit: As we've moved to python3 support in stacker, we should probably start thinking about that in stacker_blueprints as well.  One thing I know we'll need to do is stop using `iteritems` and instead use `items`.  Might as well start here :)

Reply:
done